hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -234,9 +223,9 @@ private int loadKeySet(BufferingInputStream bis, DataStructureAdapter<Data, ?> a
         return loadedKeys;
     }
 
-    private void writeKeySet(FileOutputStream fos, FileChannel outChannel, Set<KS> keySet) throws IOException {
-        for (KS key : keySet) {
-            Data dataKey = serializationService.toData(key);
+    private void writeKeySet(FileOutputStream fos, FileChannel outChannel, Iterator keySetIterator) throws IOException {
```

Comment:
I'm confused about the content of the keySetIterator. Apart from not having a type; the name gives the impression that it is an iterator that returns key-sets; not keys. But when I look at the while loop; it gives the impression that the keySetIterator is returning keys. So perhaps the name of the keySetIterator should be 'keyIterator'?